datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -557,38 +559,175 @@ pub fn coerce_int96_to_resolution(
     file_schema: &Schema,
     time_unit: &TimeUnit,
 ) -> Option<Schema> {
-    let mut transform = false;
-    let parquet_fields: HashMap<_, _> = parquet_schema
+    // Traverse the parquet_schema columns looking for int96 physical types. If encountered, insert
+    // the field's full path into a set.
+    let int96_fields: HashSet<_> = parquet_schema
```

Comment:
a nit: maybe we can do it in a single iteration like 

```
let int96_fields: HashSet<_> = parquet_schema
    .columns()
    .iter()
    .filter_map(|f| {
        (f.physical_type() == &Type::INT96).then(|| f.path().string())
    })
    .collect();
```

Reply:
Sure, didn't know that was a thing! Thank you!